### ISSUE TITLE
ERROR: can't start microservices -apps abacus-usage-meter and abacus-usage-collector

### ISSUE THREAD

**Issue by jkujovic** [cloudfoundry-incubator/cf-abacus#1035]
Hi all,
I try deploy CF-Abacus to our CF via Concourse pipeline. In job abacus-deploy and task start_abacus I get errors in start apps abacus-usage-meter and abacus-usage-collector . The error is related to AmqpConnectionManager .
Error messages is:
2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR (node:93) UnhandledPromiseRejectionWarning: Error: Must supply either `urls` or `findServers`
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at new AmqpConnectionManager (/home/vcap/app/node_modules/amqp-connection-manager/lib/AmqpConnectionManager.js:71:19)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at Object.connect (/home/vcap/app/node_modules/amqp-connection-manager/lib/index.js:15:12)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at ConnectionManager.connect (/home/vcap/app/node_modules/abacus-rabbitmq/src/lib/connection-manager.js:17:29)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at Consumer.process (/home/vcap/app/node_modules/abacus-rabbitmq/src/lib/consumer.js:76:49)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at startApp (/home/vcap/app/src/index.js:204:25)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at <anonymous>
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR     at process._tickCallback (internal/process/next_tick.js:188:7)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR (node:93) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
  2018-09-12T07:09:00.64+0000 [APP/PROC/WEB/0] ERR (node:93) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Any idea?
cf services related to rabbitmq is:
abacus-rabbitmq-instance   iwcf-rabbitmq    standard   abacus-usage-collector, abacus-usage-meter                                       create succeeded
thx for any help

**Comment by hsiliev** [comment]
Can you post the sanitized rabbitmq service credentials? 

Abacus is looking for `uri`. The one we are using looks like this:
```amqp://<user>:<password>@10.11.57.155:5672/production```

**Comment by jkujovic** [comment]
hello,

part of credential is

"protocols": {
      "amqp": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "port": 5672,
       "ssl": false,
       "uri": "amqp://user_id:password@10.0.1.32:5672/87145845-1962-4ab7-a283-c9ce95a31f57",
       "uris": [
        "amqp://user_id:password@10.0.1.32:5672/87145845-1962-4ab7-a283-c9ce95a31f57"
       ],
       "username": "user_id",
       "vhost": "87145845-1962-4ab7-a283-c9ce95a31f57"
      },

full env for app:
Getting env variables for app abacus-usage-collector in org abacus / space abacus as admin...
OK

System-Provided:
{
 "VCAP_SERVICES": {
  "my-rabbitmq": [
   {
    "binding_name": null,
    "credentials": {
     "dashboard_url": "https://cf-management-rabbitmq.mydomain.sk/#/login/user_id/password",
     "hostname": "10.0.1.32",
     "hostnames": [
      "10.0.1.32"
     ],
     "http_api_uri": "https://user_id:password@cf-management-rabbitmq.mydomain.sk/api/",
     "http_api_uris": [
      "https://user_id:password@cf-management-rabbitmq.mydomain.sk/api/"
     ],
     "password": "password",
     "protocols": {
      "amqp": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "port": 5672,
       "ssl": false,
       "uri": "amqp://user_id:password@10.0.1.32:5672/87145845-1962-4ab7-a283-c9ce95a31f57",
       "uris": [
        "amqp://user_id:password@10.0.1.32:5672/87145845-1962-4ab7-a283-c9ce95a31f57"
       ],
       "username": "user_id",
       "vhost": "87145845-1962-4ab7-a283-c9ce95a31f57"
      },
      "management": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "port": 15672,
       "ssl": false,
       "uri": "http://user_id:password@10.0.1.32:15672/87145845-1962-4ab7-a283-c9ce95a31f57",
       "uris": [
        "http://user_id:password@10.0.1.32:15672/87145845-1962-4ab7-a283-c9ce95a31f57"
       ],
       "username": "user_id",
       "vhost": "87145845-1962-4ab7-a283-c9ce95a31f57"
      },
      "management+ssl": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "path": "/api/",
       "port": 15672,
       "ssl": false,
       "uri": "http://user_id:password@10.0.1.32:15672/api/",
       "uris": [
        "http://user_id:password@10.0.1.32:15672/api/"
       ],
       "username": "user_id"
      },
      "mqtt": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "port": 1883,
       "ssl": false,
       "uri": "mqtt://87145845-1962-4ab7-a283-c9ce95a31f57%3Auser_id:password@10.0.1.32:1883",
       "uris": [
        "mqtt://87145845-1962-4ab7-a283-c9ce95a31f57%3Auser_id:password@10.0.1.32:1883"
       ],
       "username": "87145845-1962-4ab7-a283-c9ce95a31f57:user_id"
      },
      "stomp": {
       "host": "10.0.1.32",
       "hosts": [
        "10.0.1.32"
       ],
       "password": "password",
       "port": 61613,
       "ssl": false,
       "uri": "stomp://user_id:password@10.0.1.32:61613",
       "uris": [
        "stomp://user_id:password@10.0.1.32:61613"
       ],
       "username": "user_id",
       "vhost": "87145845-1962-4ab7-a283-c9ce95a31f57"
      }
     },
     "ssl": true,
     "uri": "amqps://user_id:password@10.0.1.32/87145845-1962-4ab7-a283-c9ce95a31f57",
     "uris": [
      "amqps://user_id:password@10.0.1.32/87145845-1962-4ab7-a283-c9ce95a31f57"
     ],
     "username": "user_id",
     "vhost": "87145845-1962-4ab7-a283-c9ce95a31f57"
    },
    "instance_name": "abacus-rabbitmq-instance",
    "label": "my-rabbitmq",
    "name": "abacus-rabbitmq-instance",
    "plan": "standard",
    "provider": null,
    "syslog_drain_url": null,
    "tags": [
     "rabbitmq",
     "messaging",
     "message-queue",
     "amqp",
     "stomp",
     "mqtt",
     "pivotal"
    ],
    "volume_mounts": []
   }
  ],

{
 "VCAP_APPLICATION": {
  "application_id": "a13fd9b5-ff53-48fe-b6fa-d3a4e1fccaaf",
  "application_name": "abacus-usage-collector",
  "application_uris": [
   "abacus-usage-collector.mydomain.sk"
  ],
  "application_version": "6d9efbf4-c5d6-4241-a6ed-b9bcde8b6367",
  "cf_api": "https://api.mydomain.sk",
  "limits": {
   "disk": 512,
   "fds": 16384,
   "mem": 2048
  },
  "name": "abacus-usage-collector",
  "space_id": "1b211c9b-3a78-4d63-8b51-d7331d950a05",
  "space_name": "abacus",
  "uris": [
   "abacus-usage-collector.mydomain.sk"
  ],
  "users": null,
  "version": "6d9efbf4-c5d6-4241-a6ed-b9bcde8b6367"
 }
}

User-Provided:
ACCOUNT: abacus-account-plugin
API: https://api.mydomain.sk:443
AUTH_SERVER: https://api.mydomain.sk:443
BATCH_SIZE: 10
CLIENT_ID: abacus
CLIENT_SECRET: clientsecret
CLUSTER: false
CONF: small
DBALIAS: db-collector
DBCLIENT: abacus-mongoclient
DEBUG: e-abacus-*
DEBUG_STRIPLF: true
EUREKA: abacus-eureka-plugin
EUREKA_PASSWORD: eurekapassword
EUREKA_USER: abacus-hystrix
EVAL_TIMEOUT: 100
EVAL_VMTYPE: vm2
IGNORE_ORGANIZATION: true
JWTALGO: RS256
JWTKEY: -----BEGIN PUBLIC KEY-----
jwt key
-----END PUBLIC KEY-----


MAX_INFLIGHT: 400
MAX_SYSTEM_INFLIGHT: 12
NODE_MODULES_CACHE: false
OPTIMIZE_MEMORY: true
PRIMARY_DOMAIN: mydomain.sk
PROVISIONING: abacus-provisioning-plugin
RABBIT_SERVICE_NAME: my-rabbitmq
RATE_LIMIT: {"dbalias":"db-collector","collectionName":"test-rate-limit-collection","window":{"sizeMs":60000,"maxRequestsCount":2}}
REPLAY: 1800000
SECURED: true
SKIP_SSL_VALIDATION: true
SYSTEM_INFLIGHT_ENABLED: true
THROTTLE: 100
VALIDATE_FUNCTIONS: true

No running env variables have been set

No staging env variables have been set

**Comment by hsiliev** [comment]
@jkujovic Can you please rename the service instance from  `abacus-rabbitmq-instance` instance to `rabbitmq`?

**Comment by jkujovic** [comment]
I rename it, bud still the same error

jan

**Comment by hsiliev** [comment]
Can you please post the sanitized instance details again?

**Comment by hsiliev** [comment]
With latest Abacus commits we removed service instance support as we found out this is quite unstable due to binding/unbinding can break due to update, backup or infrastructure update.

We now support plain URIs only. This will allow us to simplify the code and deployment. Service keys can still be used to obtain the URI and credentials.

The pipeline from this bug used the pipeline we open-sourced. However we were not using it internally, so it is obsolete. A replacement one should be added as part of #1057
